src/core/tsi/alts/handshaker/alts_handshaker_client.cc

```diff
@@ -536,6 +535,25 @@ static grpc_byte_buffer* get_serialized_start_client(
                                           upb_StringView_FromString(ptr->data));
     ptr = ptr->next;
   }
+  // This ensures the token string is available when the proto gets serialized.
+  std::optional<std::string> access_token = std::nullopt;
+  // Set access token if the token fetcher is available.
+  grpc::alts::TokenFetcher* token_fetcher =
+      (reinterpret_cast<grpc_alts_credentials_client_options*>(client->options))
+          ->token_fetcher.get();
+  if (token_fetcher != nullptr) {
+    absl::StatusOr<std::string> token = token_fetcher->GetToken();
+    if (!token.ok()) {
+      LOG_EVERY_N(ERROR, 1000) << "Failed to get token from the token fetcher "
```

Comment:
nit: can we use `LOG_EVERY_N_SEC(ERROR, 600)` ?

That would be more consistent with error log rates where this will be used, I think